department-of-veterans-affairs/vets-website
# 16472 update header footer injection

## Description
Some background on what we tried:
- using a loader to inject the header js into the document after the initial response. This [resulted](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/16472#issuecomment-462804339) in severe performant hit. 
- next [suggestion](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/16472#issuecomment-465143360) was to have loader file and header file both check for a query in the url. After [trying](https://github.com/department-of-veterans-affairs/vets-website/pull/9613/commits/10f01c4cdb5a1a1bc1094cd0168e087219e1f9eb) this, decided that it was a lot of extra complexity for not a lot of gain. the loader file which is intended to be nonvolatile was a very small js at the extra cost of duplicating the environment check in the loader and rewriter and maintaining a separate manifest, asset deployment, and additional call from the browser- it also would require coordinating markup changes with the teamsite admin. 
- This approach adds the ability to short circuit the header injection and to add a specific environment's assets to the document for testing purposes. performance for testing won't be production-like but this should allow us to test staging builds of the teamsite pages. 
- this should eliminate the need to use the `--local-proxy-rewrite` task for local development. with `npm run watch` running, navigating to a teamsite page with the target environment set to localhost will direct the client use `localhost` as the hostname for the teamsite assets. 

## Testing done
- run `npm run watch -- --local-proxy-rewrite`
- navigate to each 
  - http://localhost:3001/?target=https://www.va.gov/health/&targetEnvironment=vagovstaging
  - http://localhost:3001/?target=https://www.va.gov/health/&targetEnvironment=localhost
  - http://localhost:3001/?target=https://www.va.gov/health/
- verify on each what assets are downloaded and from where 
- verify on vagovstaging that the markup removes the relative path assets and uses the staging hostname



## Screenshots


## Acceptance criteria

## Definition of done
